docs/schema-changes-use-cases.md

```diff
@@ -0,0 +1,605 @@
+- Conditional deprecation: An item is deprecated based on a condition. The condition can be based on the value of another
+  item or a combination of items.
+- Generalization: An item is deprecated because it's too specific and a more general item is introduced.
+
+### Soft removal
+
+Soft removal is the process of marking an item as deprecated without removing it from the registry. Let's consider the
+following two consecutive versions of a semantic convention registry:
+
+```yaml
+# Version n
+groups:
+  - id: registry.db.metrics
+    type: attribute_group
+    stability: experimental
+    attributes:
+      - id: db.jdbc.driver_classname
+        type: string
+        stability: experimental
+```
+
+To deprecate the `db.jdbc.driver_classname` attribute, the following changes are made in the
+next version of the registry:
+
+```yaml
+# Version n+1
+groups:
+  - id: registry.db.metrics.deprecated
+    type: attribute_group
+    stability: experimental
+    attributes:
+      - id: db.jdbc.driver_classname
+        type: string
+        brief: 'Removed, no replacement at this time.'
+        deprecated:
+          type: deprecated
+        stability: experimental
+```
+
+From these changes, Weaver can generate a diff report that shows the deprecation of the
+`db.jdbc.driver_classname` attribute.
+
+```yaml
+head:
+  semconv_version: n
+baseline:
+  semconv_version: n+1
+changes:
+  attributes:
+    - name: db.jdbc.driver_classname
+      type: deprecated
+      note: Removed, no replacement at this time.
+```
+
+### Conditional deprecation
+
+Sometimes, the deprecation of an item is context-dependent. Let's consider the following two
+consecutive versions of a semantic convention registry:
+
+```yaml
+# Version n
+groups:
+  - id: registry.db
+    type: attribute_group
+    stability: experimental
+    attributes:
+      - id: db.instance.id
+        type: string
+        stability: experimental
+```
+
+To deprecate the `db.instance.id` attribute based on the value of the `db.system` attribute,
+the following changes are made in the next version of the registry. Note that we suppose in this example that
+the `db.system` attribute is already defined in the registry.
+
+```yaml
+# Version n+1
+groups:
+  - id: registry.db.deprecated
+    type: attribute_group
+    stability: experimental
+    attributes:
+      - id: db.instance.id
+        type: string
+        brief: 'Deprecated, no general replacement at this time. For Elasticsearch, use `db.elasticsearch.node.name` instead.'
+        deprecated:
+          type: conditionally_deprecated
+          forward: >
+            if attributes['db.system'] == 'elasticsearch' then attributes['db.elasticsearch.node.name'] = attributes['db.instance.id']
+            else drop attributes['db.instance.id']
+          backward: >
+            if attributes['db.system'] == 'elasticsearch' then attributes['db.instance.id'] = attributes['db.elasticsearch.node.name']
+        stability: experimental
+```
```

Comment:
Like my previous comment. When the instructions involve other attributes then the change is dependent on the presence of those attributes in that signal definition. Perhaps defining the instructions on the attribute_group means it's universal for all uses via references? Maybe this can be overridden with further instructions on the span, for example, to deviate from this default?

Reply:
Yes, in the `attribute_group`, the setting is intended to be universal. However, I agree that we should demonstrate how this setting can be overridden when dealing with a specific signal.

Reply:
See my previous comment.